recipes/mongo-cxx-driver/all/conanfile.py

```diff
@@ -0,0 +1,141 @@
+        self.requires("mongo-c-driver/1.17.2")
+        if self.options.polyfill == "boost":
+            self.requires("boost/1.74.0")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-r" + self.version, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["BSONCXX_POLY_USE_MNMLSTC"] = self.options.polyfill == "mnmlstc"
+        self._cmake.definitions["BSONCXX_POLY_USE_STD"] = self.options.polyfill == "std"
+        self._cmake.definitions["BSONCXX_POLY_USE_STD_EXPERIMENTAL"] = self.options.polyfill == "experimental"
+        self._cmake.definitions["BSONCXX_POLY_USE_BOOST"] = self.options.polyfill == "boost"
+        self._cmake.definitions["BUILD_VERSION"] = self.version
+        self._cmake.definitions["BSONCXX_LINK_WITH_STATIC_MONGOC"] = not self.options["mongo-c-driver"].shared
+        self._cmake.definitions["MONGOCXX_LINK_WITH_STATIC_MONGOC"] = not self.options["mongo-c-driver"].shared
+        # FIXME: two CMake module/config files should be generated (mongoc-1.0-config.cmake and bson-1.0-config.cmake),
+        # but it can't be modeled right now.
+        # Fix should happen in mongo-c-driver recipe
+        if not os.path.exists("Findbson-1.0.cmake"):
+            self.output.info("Copying mongoc config file to bson")
+            shutil.copy("Findmongoc-1.0.cmake", "Findbson-1.0.cmake")
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+    def build(self):
+        self._patch_sources()
+
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        self.copy(pattern="THIRD-PARTY-NOTICES", dst="licenses", src=self._source_subfolder)
+
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        if self.settings.os == "Windows":
+            for dll_file in glob.glob(os.path.join(self.package_folder, "bin", "*.dll")):
+                if os.path.basename(dll_file).startswith(("concrt", "msvcp", "vcruntime")):
+                    os.remove(dll_file)
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        # FIXME: two CMake module/config files should be generated (mongocxx-config.cmake and bsoncxx-config.cmake),
+        # but it can't be modeled right now.
+        self.cpp_info.filenames["cmake_find_package"] = "mongocxx"
+        self.cpp_info.filenames["cmake_find_package_multi"] = "mongocxx"
+        self.cpp_info.names["cmake_find_package"] = "mongo"
```

Comment:
```mongo-c-driver contains same value```, is it ok?

Reply:
Yes, they share the same CMake namespace.

Reply:
Here is an example:
https://github.com/mongodb/mongo-cxx-driver/blob/5efb920e0ee9f671c0b7495c26a97b37c0ca6c4e/examples/projects/mongocxx/cmake/static/CMakeLists.txt#L60